src/main/java/com/iexec/core/replicate/ReplicateModel.java

```diff
@@ -33,22 +36,45 @@ public class ReplicateModel {
     private String chainTaskId;
     private String walletAddress;
     private ReplicateStatus currentStatus;
-    //TODO: Move/extract details here instead of encapsulating them within status updates
-    private List<ReplicateStatusUpdate> statusUpdateList;
+    private List<ReplicateStatusUpdateModel> statusUpdateList;
     private String resultLink;
     private String chainCallbackData;
     private String contributionHash;
-    private String appStdout;
+    private String appLogs;
+    private Integer appExitCode; //null means unset
+    private String teeSessionGenerationError; // null means unset
 
     public static ReplicateModel fromEntity(Replicate entity) {
+        final List<ReplicateStatusUpdateModel> statusUpdateList = new ArrayList<>();
+
+        Integer appExitCode = null;
+        String teeSessionGenerationError = null;
+        for (ReplicateStatusUpdate replicateStatusUpdate : entity.getStatusUpdateList()) {
+            statusUpdateList.add(ReplicateStatusUpdateModel.fromEntity(replicateStatusUpdate));
+            ReplicateStatusDetails details = replicateStatusUpdate.getDetails();
+            if (details != null) {
+                if (appExitCode == null) {
```

Comment:
A little awkward, but I think I see the issue you have
Maybe something like
```
if(this.exitCode != null && details.exitCode != null) // trying to write twice
  throw new MutipleOccurencesOfFielNotAllowed()
}
```
?

Reply:
Either we check `appExitCode` is null and we can assign at each iteration whatever the value of `details.getExitCode()` is, either we check `details.getExitCode()` is non-null and we assign only then. That's why this loop looks strange.

Your proposal is great and should be introduced but it does not solve my initial issue.